File: mobile_client/app/src/main/java/wpam/mobile_client/protocol/PlotBuilder.java
package wpam.mobile_client.protocol;

import java.io.Closeable;
import java.util.ArrayList;

import wpam.mobile_client.sensor_tag.ParameterType;
import wpam.mobile_client.sensor_tag.SensorTagType;

public final class PlotBuilder implements CommandBuilder, Closeable
{
    public PlotBuilder()
    {}
    public final void close()
    {}

    @Override
    public Command create(ArrayList<Integer> commandInBytes)
    {
        ParameterType type = ParameterType.forValue(commandInBytes.get(initialDataPosition));
        SensorTagType sensorType = SensorTagType.forValue(commandInBytes.get(initialDataPosition+1));

        Command response  = new PlotCommand(type, sensorType);
        return response;
    }
}

File: server/src/protocol/PlotBuilder.cpp
#include "PlotBuilder.h"
#include "PlotCommand.h"

using namespace std;
using namespace communication;

PlotBuilder::PlotBuilder()
{}

PlotBuilder::~PlotBuilder()
{}

unique_ptr<Command> PlotBuilder::create(const vector<uint8_t> &commandInBytes)
{
    const auto parameterType = static_cast<ParameterType>(commandInBytes[initialDataPosition]);
    const auto sensorTagType = static_cast<SensorTagType>(commandInBytes[initialDataPosition+1]);

    auto command = make_unique<PlotCommand>(parameterType, sensorTagType);

    return move(command);
}
